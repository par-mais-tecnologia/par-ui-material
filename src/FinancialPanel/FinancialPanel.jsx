import React, { Component } from 'react'
import { withStyles } from '@material-ui/core/styles'
import ExpansionPanel from '@material-ui/core/ExpansionPanel'
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails'
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary'
import Typography from '@material-ui/core/Typography'
import ExpandMoreIcon from '@material-ui/icons/ExpandMore'
import { getCurrencyFormat } from '../Core/masks'

const styles = theme => (theme.financialPanel)

class FinancialPanel extends Component {
  state = {
    hidden: true
  }

  handleTrigger = () => (
    this.setState({
      hidden: !this.state.hidden
    })
  )

  render () {
    const { classes, data } = this.props

    return (
      <div className={classes.root}>

        <ExpansionPanel
          onChange={() => this.handleTrigger()}
          classes={{ root: classes.header }}>
          <ExpansionPanelSummary
            expandIcon={<ExpandMoreIcon />}>
            <div className={classes.flexBoxStart}>
              <div className={classes.box}
                style={{ backgroundColor: this.props.color }} />
              <Typography
                variant='title'
                className={classes.value}>
                {this.props.headerPercentage * 100}%
              </Typography>
              <Typography
                variant='title'
                className={classes.name}
                style={{ color: this.props.color }}
              >
                {this.props.headerName}
              </Typography>
            </div>
          </ExpansionPanelSummary>
        </ExpansionPanel>
        <div className={this.state.hidden ? classes.close : classes.open}>

          {
            data.map((obj) => (

              <ExpansionPanel hidden={this.state.hidden}>
                <ExpansionPanelSummary
                  expandIcon={<ExpandMoreIcon />}>
                  <div className={classes.flexBoxBetween}>
                    <Typography
                      variant='title'
                      className={classes.name}>
                      {obj.name}
                    </Typography>
                    <Typography
                      variant='title'
                      className={classes.walletComposition}>
                      {obj.walletComposition * 100}%
                    </Typography>
                  </div>
                </ExpansionPanelSummary>

                <ExpansionPanelDetails>
                  <div className={classes.detailsLine}>
                    <Typography className={classes.detailsTitle}>
                        Rentabilidade dos Ãºltimos 24 meses
                    </Typography>
                    <Typography className={classes.detailsContent}>
                      {obj.cdi * 100}% do {obj.index}
                    </Typography>
                  </div>

                  <div className={classes.detailsLine}>
                    <Typography className={classes.detailsTitle}>
                        Carteira
                    </Typography>
                    <Typography className={classes.detailsContent}>
                      {obj.portfolio}
                    </Typography>
                  </div>

                  <div className={classes.detailsLine}>
                    <Typography className={classes.detailsTitle}>
                        Pl atual
                    </Typography>
                    <Typography className={classes.detailsContent}>
                        R$ {getCurrencyFormat(obj.gross)}
                    </Typography>
                  </div>

                  <div className={classes.detailsLine}>
                    <Typography className={classes.detailsTitle}>
                        Cnpj do fundo
                    </Typography>
                    <Typography className={classes.detailsContent}>
                      {obj.cnpj}
                    </Typography>
                  </div>
                </ExpansionPanelDetails>

              </ExpansionPanel>
            ))
          }
        </div>
      </div>
    )
  }
}

export default withStyles(styles)(FinancialPanel)
