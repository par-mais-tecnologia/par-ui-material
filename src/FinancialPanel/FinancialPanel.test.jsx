import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import FinancialPanel from './FinancialPanel'

configure({ adapter: new Adapter() })

export const test = describe('FinancialPanel', () => {
  const getShallow = () => (
    <FinancialPanel
      headerName='Multimercado'
      headerPercentage='0.35'
      color='#0580C4'
      data={{
        name: 'AZ Legan Low VOL',
        walletComposition: 0.2,
        performance24: 1.38,
        portfolio: 'Ações, moedas, mercado de juros',
        index: 'CDI',
        gross: 297321988,
        cnpj: '05.488.919/001-90'
      }} />
  )

  it('shallow FinancialPanel', () => {
    const wrapper = shallow(
      <FinancialPanel />
    )
    expect(wrapper).toBeTruthy()
  })

  it('shallow FinancialPanel match Snapshot', () => {
    const wrapper = shallow(getShallow())
    expect(wrapper).toMatchSnapshot()
  })
})
