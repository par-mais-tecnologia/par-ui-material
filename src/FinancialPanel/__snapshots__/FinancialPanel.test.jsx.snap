// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinancialPanel shallow FinancialPanel match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(FinancialPanel)
    color="#0580C4"
    data={
      Object {
        "cnpj": "05.488.919/001-90",
        "gross": 297321988,
        "index": "CDI",
        "name": "AZ Legan Low VOL",
        "performance24": 1.38,
        "portfolio": "Ações, moedas, mercado de juros",
        "walletComposition": 0.2,
      }
    }
    headerName="Multimercado"
    headerPercentage="0.35"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {},
      "color": "#0580C4",
      "data": Object {
        "cnpj": "05.488.919/001-90",
        "gross": 297321988,
        "index": "CDI",
        "name": "AZ Legan Low VOL",
        "performance24": 1.38,
        "portfolio": "Ações, moedas, mercado de juros",
        "walletComposition": 0.2,
      },
      "headerName": "Multimercado",
      "headerPercentage": "0.35",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {},
        "color": "#0580C4",
        "data": Object {
          "cnpj": "05.488.919/001-90",
          "gross": 297321988,
          "index": "CDI",
          "name": "AZ Legan Low VOL",
          "performance24": 1.38,
          "portfolio": "Ações, moedas, mercado de juros",
          "walletComposition": 0.2,
        },
        "headerName": "Multimercado",
        "headerPercentage": "0.35",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
